server/index.js

```diff
@@ -316,6 +325,305 @@ app.get('/api/fighters/data/:type', (req, res, next) => {
+          .then(result => {
+            Object.assign(fullResult, result.rows[0]);
+            handleMovements(counter + 1)
+          })
+            .catch(err => {
+              if (/already exists/g.test(err.detail)
+                || /not present/g.test(err.detail)) {
+                throw new ClientError(400, `${err.detail}`);
+              } else {
+                next(err);
+              }
+            });
+        } else {
+          const sql2 = `
+          INSERT INTO public.dodging
+            ("activeFrames", "totalFrames")
+          VALUES ($1, $2)
+          RETURNING *;
+          `;
+          const params2 = [activeFrames, totalFrames];
+          return db.query(sql2, params2)
+            .then(result => {
+              Object.assign(fullResult, result.rows[0]);
+              res.status(201).json(fullResult)
+            })
+            .catch(err => {
+              if (/already exists/g.test(err.detail)
+                || /not present/g.test(err.detail)) {
+                throw new ClientError(400, `${err.detail}`);
+              } else {
+                next(err);
+              }
+            });
+        }
+      }
+    }
+});
+
+app.post('/api/add/stats', (req, res, next) => {
+  const { name, statValue } = req.body;
+  let { fighterId } = req.body;
+  fighterId = Number(fighterId);
+  const fullResult = {};
+  if (!fighterId) {
+    throw new ClientError(400, 'fighterId must be a number');
+    return;
+  }
+  if (!checkValidity([name, statValue])) {
+    throw new ClientError(400, 'must have (fighterId), (name), and (statValue) as parameters');
+    return;
+  }
+  if (typeof fighterId === 'number'
+    & typeof name === 'string'
+    & typeof statValue === 'string') {
+    return handleStats(1)
+    function handleStats(counter) {
+      if (counter === 1) {
+        const sql = `
+          INSERT INTO public.stats
+            ("fighterId", "name", "type")
+          VALUES ($1, $2, 'stat')
+          RETURNING *;
+          `;
+        const params = [fighterId, name]
+        return db.query(sql, params)
+          .then(result => {
+            Object.assign(fullResult, result.rows[0]);
+            handleStats(counter + 1)
+          })
+          .catch(err => {
+            if (/already exists/g.test(err.detail)
+              || /not present/g.test(err.detail)) {
+              throw new ClientError(400, `${err.detail}`);
+            } else {
+              next(err);
+            }
+          });
+      } else {
+        const sql2 = `
+          INSERT INTO public.miscellaneous
+            ("statValue")
+          VALUES ($1)
+          RETURNING *;
+          `;
+        const params2 = [statValue];
+        return db.query(sql2, params2)
+          .then(result => {
+            Object.assign(fullResult, result.rows[0]);
+            res.status(201).json(fullResult)
+          })
+          .catch(err => {
+            if (/already exists/g.test(err.detail)
```

Comment:
Instead of regex testing error messages (which is very fragile and relies on the error messages to never change), use what's called an "upsert", which is a feature of postgres. You can use an `on conflict` to `do nothing` in your SQL code.

https://www.postgresql.org/docs/current/sql-insert.html#SQL-ON-CONFLICT